compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -13807,6 +13826,12 @@ private STNode parseTypedBindingPatternOrExprRhs(SyntaxKind nextTokenKind, STNod
                     return solution.recoveredNode;
                 }
 
+                // If the recovered token is not something that can be re-parsed,
+                // then don't try to re-parse the same rule.
+                if (solution.tokenKind == SyntaxKind.NONE) {
+                    return typeOrExpr;
+                }
```

Comment:
Can you explain why do we need to check `SyntaxKind.NONE` here?

Reply:
because recovery returns it when we try to parse source 2 from https://github.com/ballerina-platform/ballerina-lang/issues/24651

Reply:
Why is it returning NONE? Shouldn't we fix that?